internal/commands/show/show.go

```diff
@@ -0,0 +1,214 @@
+			Parents: parents,
+			Author: &predicate.Author{
+				Name:  commit.Author.Name,
+				Email: commit.Author.Email,
+				Date:  commit.Author.When,
+			},
+			Committer: &predicate.Author{
+				Name:  commit.Committer.Name,
+				Email: commit.Committer.Email,
+				Date:  commit.Committer.When,
+			},
+			Message: commit.Message,
+		},
+		Signature: commit.PGPSignature,
+	}
+
+	// If we can a PEM encoded signature, try and extract certificate details.
+	pem, _ := pem.Decode([]byte(commit.PGPSignature))
+	if pem != nil {
+		sigs, err := parseSignature(pem.Bytes)
+		if err != nil {
+			return nil, err
+		}
+		predicate.SignerInfo = sigs
+	}
+
+	// Try and resolve the remote name to use as the subject name.
+	// If the repo does not have a remote configured then this will be left
+	// blank.
+	resolvedRemote, err := repo.Remote(remote)
+	if err != nil && !errors.Is(err, git.ErrRemoteNotFound) {
+		return nil, err
+	}
+	remoteName := ""
+	if resolvedRemote != nil && len(resolvedRemote.Config().URLs) > 0 {
+		remoteName = resolvedRemote.Config().URLs[0]
+	}
+
+	// Wrap predicate in in-toto Statement.
+	return &in_toto.Statement{
+		StatementHeader: in_toto.StatementHeader{
+			Type: in_toto.StatementInTotoV01,
+			Subject: []in_toto.Subject{{
+				Name: remoteName,
+				Digest: v02.DigestSet{
+					// TODO?: Figure out if/how to support git sha256 - this
+					// will likely depend on upstream support in go-git.
+					// See https://github.com/go-git/go-git/issues/229.
+					"sha1": hash.String(),
+				},
+			}},
+			PredicateType: predicateType,
+		},
+		Predicate: predicate,
+	}, nil
+}
+
+func parseSignature(raw []byte) ([]*predicate.SignerInfo, error) {
+	ci, err := protocol.ParseContentInfo(raw)
+	if err != nil {
+		return nil, err
+	}
+
+	sd, err := ci.SignedDataContent()
+	if err != nil {
+		return nil, err
+	}
+
+	certs, err := sd.X509Certificates()
+	if err != nil {
+		return nil, err
+	}
+
+	// A signature may have multiple signers associated to it -
+	// extract each SignerInfo separately.
+	out := make([]*predicate.SignerInfo, 0, len(sd.SignerInfos))
+	for _, si := range sd.SignerInfos {
+		cert, err := si.FindCertificate(certs)
+		if err != nil {
+			continue
+		}
+		b, err := cryptoutils.MarshalCertificateToPEM(cert)
+		if err != nil {
+			return nil, err
+		}
+		sa, err := si.SignedAttrs.MarshaledForVerification()
+		if err != nil {
+			return nil, err
+		}
+		out = append(out, &predicate.SignerInfo{
+			Certificate: string(b),
+			Attributes:  base64.StdEncoding.EncodeToString(sa),
+		})
+	}
+
+	return out, nil
+}
+
+func New(cfg *config.Config) *cobra.Command {
+	o := &options{}
+
+	cmd := &cobra.Command{
+		Use:   "show [revision]",
+		Short: "Show source predicate information",
+		Long: `Show source predicate information
+
+Prints an in-toto style predicate for the specified revision.
+If no revision is specified, HEAD is used.
+
+This command is experimental, and its CLI surface may change.`,
+		Args: cobra.NoArgs,
```

Comment:
This will break if you pass a revision

Reply:
Yeah this was accidentally left in place when I was trying to debug the tty weirdness. Removed.